documentation/src/test/java/example/exception/ExceptionAssertionExactDemo.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package example.exception;
+
+import static org.junit.jupiter.api.Assertions.assertThrowsExactly;
+
+import org.junit.jupiter.api.Test;
+
+// @formatter:off
+// tag::user_guide[]
+
+public class ExceptionAssertionExactDemo {
+
+    // end::user_guide[]
+    @extensions.ExpectToFail
+    // tag::user_guide[]
+    @Test
+    void testExpectedExceptionIsThrown() {
+        assertThrowsExactly(IllegalArgumentException.class, () -> {
+            throw new IllegalArgumentException("Expected exception");
+        }, "This will fail because because we expect IllegalArgumentException to be thrown");
```

Comment:
This message looks incorrect? The test will actually pass because exactly the same exception type is thrown.